Jetcaster/wear/src/main/java/com/example/jetcaster/ui/library/QueueScreen.kt

```diff
@@ -0,0 +1,199 @@
+        onErrorDialogCancelClick = onErrorDialogCancelClick,
+        onPlayButtonClick = onPlayButtonClick,
+        modifier = modifier,
+    )
+}
+
+@Composable
+fun PodcastDetailsScreen(
+    viewState: QueueScreenState,
+    onChangeSpeedButtonClick: () -> Unit,
+    onPlayButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    onEpisodeItemClick: (EpisodeToPodcast) -> Unit,
+    onPlayEpisode: (PlayerEpisode) -> Unit,
+    onErrorDialogCancelClick: () -> Unit
+) {
+    val columnState = rememberResponsiveColumnState(
+        contentPadding = padding(
+            first = ScalingLazyColumnDefaults.ItemType.Text,
+            last = ScalingLazyColumnDefaults.ItemType.Chip
+        )
+    )
+    ScreenScaffold(
+        scrollState = columnState,
+        modifier = modifier
+    ) {
+        when (viewState) {
+            is QueueScreenState.Loaded -> {
+                EntityScreen(
+                    columnState = columnState,
+                    headerContent = {
+                        DefaultEntityScreenHeader(
+                            title = stringResource(R.string.queue)
+                        )
+                    },
+                    buttonsContent = {
+                        ButtonsContent(
+                            episodes = viewState.episodeList,
+                            onChangeSpeedButtonClick = onChangeSpeedButtonClick,
+                            onPlayButtonClick = onPlayButtonClick,
+                            onPlayEpisode = onPlayEpisode
+                        )
+                    },
+                    content = {
+                        items(count = viewState.episodeList.size) {
+                                index ->
+                            MediaContent(
+                                episode = viewState.episodeList[index],
+                                episodeArtworkPlaceholder = rememberVectorPainter(
+                                    image = Icons.Default.MusicNote,
+                                    tintColor = Color.Blue,
+                                ),
+                                onEpisodeItemClick
+                            )
+                        }
+                    }
+                )
+            }
+            QueueScreenState.Loading -> {
+                LoadingEntityScreen(columnState)
+            }
+            QueueScreenState.Error -> {
+                AlertDialog(
+                    showDialog = true,
+                    onDismiss = onErrorDialogCancelClick,
+                    title = stringResource(R.string.display_nothing_in_queue),
+                    message = stringResource(R.string.failed_loading_episodes_from_queue)
+                )
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalHorologistApi::class)
+@Composable
+fun ButtonsContent(
+    episodes: List<PlayerEpisode>,
+    onChangeSpeedButtonClick: () -> Unit,
+    onPlayButtonClick: () -> Unit,
```

Comment:
For this component, I'd probably merge onPlayButtonClick and onPlayButton. It's a bit confusing why there are two, which is clearer in the composable with both a view model and navigation.